src/sql/workbench/contrib/commandLine/electron-sandbox/commandLine.ts

```diff
@@ -118,37 +118,27 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 			});
 			return;
 		}
-		let connectedContext: azdata.ConnectedContext = undefined;
-		if (profile) {
-			if (profile.providerName && !this._capabilitiesService.providers[profile.providerName]) {
-				const installed = await this._connectionManagementService.handleUnsupportedProvider(profile.providerName);
-				if (!installed) {
-					// User cancelled install prompt so exit early since we won't be able to connect
-					return;
+		if (commandName) {
+			if (profile && commandName === Command.connect) {
+				if (this._notificationService) {
+					this._notificationService.status(localize('connectingLabel', "Connecting: {0}", profile.serverName), { hideAfter: 2500 });
+				}
+				try {
+					await this._connectionManagementService.connectIfNotConnected(profile, args.showDashboard ? 'dashboard' : 'connection', true);
+					// Before sending to extensions, we should a) serialize to IConnectionProfile or things will fail,
+					// and b) use the latest version of the profile from the service so most fields are filled in.
+					let updatedProfile = this._connectionManagementService.getConnectionProfileById(profile.id);
+					connectedContext = { connectionProfile: new ConnectionProfile(this._capabilitiesService, updatedProfile).toIConnectionProfile() };
+				} catch (err) {
+					this.logService.warn('Failed to connect due to error: ' + getErrorMessage(err));
 				}
-				// Recreate the profile here now that we have our provider registered so that we get the correct
-				// option names. See https://github.com/microsoft/azuredatastudio/issues/20773 for details about the issue
```

Comment:
@Charles-Gagnon 
To resolve this, I'm letting the provider be handled first, before profile is created. Let me know if I'm missing something..

Reply:
Seems like a good idea to me! And yeah, that should fix this issue I believe